Agent/Utilities/NRLogger.m

```diff
@@ -299,19 +316,37 @@ - (NSData*) jsonDictionary:(NSDictionary*)message {
         entityGuid = @"";
     }
 
-    NSDictionary *requiredAttributes = @{NRLogMessageLevelKey:      [message objectForKey:NRLogMessageLevelKey],                                                                 // 1
-                                         NRLogMessageFileKey:       [[message objectForKey:NRLogMessageFileKey]stringByReplacingOccurrencesOfString:@"\"" withString:@"\\\""],   // 2
-                                         NRLogMessageLineNumberKey: [message objectForKey:NRLogMessageLineNumberKey],                                                            // 3
-                                         NRLogMessageMethodKey:     [[message objectForKey:NRLogMessageMethodKey]stringByReplacingOccurrencesOfString:@"\"" withString:@"\\\""], // 4
-                                         NRLogMessageTimestampKey:  [message objectForKey:NRLogMessageTimestampKey],                                                             // 5
-                                         NRLogMessageMessageKey:    [[message objectForKey:NRLogMessageMessageKey]stringByReplacingOccurrencesOfString:@"\"" withString:@"\\\""],// 6
-                                         NRLogMessageSessionIdKey: NRSessionId,                                                                                                  // 7
-                                         NRLogMessageAppIdKey: nrAppId,                                                                                                          // 8
-                                         NRLogMessageEntityGuidKey: entityGuid,                                                                                                  // 9
-                                         NRLogMessageInstrumentationProviderKey: NRLogMessageMobileValue,                                                                        // 10
-                                         NRLogMessageInstrumentationNameKey: nativePlatform,                                                                                               // 11
-                                         NRLogMessageInstrumentationVersionKey: [NRMAAgentConfiguration connectionInformation].deviceInformation.agentVersion,                   // 12
-                                         NRLogMessageInstrumentationCollectorKey: nativePlatform};                                                                                         // 13
+    NSMutableDictionary *requiredAttributes = [NSMutableDictionary dictionary];
```

Comment:
Why was this changed? I am working on some changes in this area. Im concerned that we might miss required attributes with the way this was changed.

Reply:
If you'd like to keep the changes, which is could be okay if they aren't required by log API... then lets make the following change:

```
                                         // "instrumentation.name"
                                         NRLogMessageInstrumentationNameKey: name, // 11                                                         
                                         // "collector.name
                                         NRLogMessageInstrumentationCollectorKey: nativePlatform};  // 13   
```

Reply:
I made this change so that the agent wouldn't crash if the value was null. There is still probably a change we could make so that it again won't add a log if it doesn't have the required attributes.